SimPEG/Examples/SEIS_TomoJointWithVolume.py

```diff
@@ -0,0 +1,203 @@
+              (uses the effective medium theory mapping)
+            - model regularization
+
+    """
+    nC = 40
+    de = 1.
+    h = np.ones(nC)*de/nC
+    M = Mesh.TensorMesh([h, h])
+
+    y = np.linspace(M.vectorCCy[0], M.vectorCCx[-1], int(np.floor(nC/4)))
+    rlocs = np.c_[0*y+M.vectorCCx[-1], y]
+    rx = StraightRay.Rx(rlocs, None)
+
+    srcList = [
+        StraightRay.Src(loc=np.r_[M.vectorCCx[0], yi], rxList=[rx]) for yi in y
+    ]
+
+    # phi model
+    phi0 = 0
+    phi1 = 0.65
+    phitrue = Utils.ModelBuilder.defineBlock(
+        M.gridCC, [0.4, 0.6], [0.6, 0.4], [phi1, phi0]
+    )
+
+    knownVolume = np.sum(phitrue*M.vol)
+    print('True Volume: {}'.format(knownVolume))
+
+    # Set up true conductivity model and plot the model transform
+    sigma0 = np.exp(1)
+    sigma1 = 1e4
+
+    if plotIt:
+        fig, ax = plt.subplots(1, 1)
+        sigmaMapTest = Maps.SelfConsistentEffectiveMedium(
+            nP=1000, sigma0=sigma0, sigma1=sigma1, rel_tol=1e-1, maxIter=150
+        )
+        testphis = np.linspace(0., 1., 1000)
+
+        sigetest = sigmaMapTest * testphis
+        ax.semilogy(testphis, sigetest)
+        ax.set_title('Model Transform')
+        ax.set_xlabel('$\\varphi$')
+        ax.set_ylabel('$\sigma$')
+
+    sigmaMap = Maps.SelfConsistentEffectiveMedium(
+        M, sigma0=sigma0, sigma1=sigma1
+    )
+
+    # scale the slowness so it is on a ~linear scale
+    slownessMap = Maps.LogMap(M) * sigmaMap
+
+    # set up the true sig model and log model dobs
+    sigtrue = sigmaMap * phitrue
+
+    # modt = Model.BaseModel(M);
+    slownesstrue = slownessMap * phitrue # true model (m = log(sigma))
+
+    # set up the problem and survey
+    survey = StraightRay.Survey(srcList)
+    problem = StraightRay.Problem(M, slownessMap=slownessMap)
+    problem.pair(survey)
+
+    if plotIt:
+        fig, ax = plt.subplots(1, 1)
+        cb = plt.colorbar(M.plotImage(phitrue, ax=ax)[0], ax=ax)
+        survey.plot(ax=ax)
+        cb.set_label('$\\varphi$')
+
+    # get observed data
+    dobs = survey.makeSyntheticData(phitrue, std=0.03, force=True)
+    dpred = survey.dpred(np.zeros(M.nC))
+
+    # objective function pieces
+    reg = Regularization.Tikhonov(M)
+    dmis = DataMisfit.l2_DataMisfit(survey)
+    dmisVol = Volume(mesh=M, knownVolume=knownVolume)
+    beta = 0.25
+    maxIter = 15
+
+    # without the volume regularization
+    opt = Optimization.ProjectedGNCG(maxIter=maxIter, lower=0.0, upper=1.0)
+    opt.remember('xc')
+    invProb = InvProblem.BaseInvProblem(dmis, reg, opt, beta=beta)
+    inv = Inversion.BaseInversion(invProb)
+
+    mopt1 = inv.run(np.zeros(M.nC)+1e-16)
+    print(
+        '\nTotal recovered volume (no vol misfit term in inversion): '
+        '{}'.format(
+            dmisVol(mopt1)
+        )
+    )
+
+    # with the volume regularization
+    vol_multiplier = 9e4
+    reg2 = reg
+    dmis2 = dmis + vol_multiplier * dmisVol
```

Comment:
Wouldn't it be nice if you could just easily create data misfits by adding them together?

Oh wait!